src/logic/tokens/api/fetchToken.ts

```diff
@@ -1,16 +1,31 @@
-import axios from 'axios'
+import axios, { AxiosResponse } from 'axios'
 
 import { getRelayUrl } from 'src/config/index'
 
-const fetchToken = (tokenAddress) => {
+type TokenData = {
+  type: string
+  address: string
+  name: string
+  symbol: string
+  decimals: number
+  logoUri: string
+}
+
+interface TokenInfoResponseData extends AxiosResponse {
+  data: TokenData
+}
+
+const fetchToken = async (tokenAddress: string): Promise<TokenData | null> => {
   const apiUrl = getRelayUrl()
```

Comment:
lets use /tokens endpoint from safe-transaction-service

Reply:
That endpoint give us all the tokens from the backend without `tokenAddress ` as param. This is better because just give us what we need

Reply:
So the endpoint for an individual token exists and could be found at swagger, lets postpone the migration though - the response `/tokens` from transaction service is polymorphic as it includes erc20, erc721 and stuff and the migration is a little bit more harder in this case

Reply:
I will create a follow-up ticket